src/state_control/state_control_Lara.cpp

```diff
@@ -3211,366 +3217,855 @@ static void StateControl_SetWeaponMeshOn(ss_bone_frame_p bf, skeletal_model_p sm
+                        t = (range * i) / num_shots;
+                        vec3_add_mul(to, from, dir, t);
+                        t = 8.0f * i;
+                        switch (i % 4)
+                        {
+                        case 0: vec3_add_mul(to, to, tr + 0, t); break;
+                        case 1: vec3_add_mul(to, to, tr + 4, t); break;
+                        case 2: vec3_add_mul(to, to, tr + 0, -t); break;
+                        case 3: vec3_add_mul(to, to, tr + 4, -t); break;
+                        }
+                    }
+                }
+                else if (target)
+                {
+                    Anim_SetAnimation(ss_anim, 0, -1);
+                }
+                else
+                {
+                    Anim_SetAnimation(ss_anim, 4, 0);
+                }
+            }
+            if ((ss_anim->frame_changing_state == 0x01) && (ss_anim->prev_frame == 2) && Sounds[RELOAD])
+            {
+                Audio_Send(Sounds[RELOAD], TR_AUDIO_EMITTER_ENTITY, ent->id);
+            }
+            break;
+
+        case 3: // idle - > hide;
+            if (Anim_IncTime(ss_anim, time))
+            {
+                SSBoneFrame_DisableOverrideAnim(ent->bf, ss_anim);
+            }
+            if ((ss_anim->frame_changing_state >= 0x01) && (ss_anim->prev_frame == 23))
+            {
+                if (ver >= TR_II)
+                {
+                    StateControl_SetWeaponMeshOn(ent->bf, ss_anim->model, 14);
+                }
+                else
+                {
+                    StateControl_SetWeaponMeshOn(ent->bf, ss_anim->model, 7);
+                }
+
+                StateControl_SetWeaponMeshOff(ent->bf, 10);
+
+                if (Sounds[HIDE])
+                {
+                    Audio_Send(Sounds[HIDE], TR_AUDIO_EMITTER_ENTITY, ent->id);
+                }
+            }
+            break;
+
+        case 4: // aim - > idle;
+            /// Shoot sound forced stopping because LOOP mode (only loop mode) ! and echo sound here !
+            if ((ss_anim->current_frame == 1))
+            {
+                // delete sound (loop mode killer)
+                Audio_Kill(Sounds[SHOT], TR_AUDIO_EMITTER_ENTITY, ent->id);
+
+                // echo sound
+                if ((ver == TR_II || ver == TR_II_DEMO) && (ss_anim->model->id == TR2_M16) && (ss_anim->anim_frame_flags == ANIM_FRAME_REVERSE))
+                {
+                    Audio_Send(Sounds[ECHO], TR_AUDIO_EMITTER_ENTITY, ent->id);
+                }
+            }
+
+            if (Anim_IncTime(ss_anim, time))
+            {
+                Anim_SetAnimation(ss_anim, 0, 0);
+            }
+
+            break;
+        };
+    }
+
     return ss_anim->frame_changing_state;
 }
 
-
-int StateControl_LaraDoTwoHandWeponFrame(struct entity_s *ent, struct  ss_animation_s *ss_anim, float time)
+/// LaraGetWeaponSFX(ss_anim, ver)
+int *LaraGetWeaponSFX(struct ss_animation_s *ss_anim, int32_t ver)
```

Comment:
static int * is not good here + very short defines (may be collisions)
it is better and more readable to use local type like
struct weapon_sound_inf_s
{
    int16_t shoot;
    int16_t draw;
    .....
}

and 

struct weapon_sound_inf_s LaraGetWeaponSFX(struct ss_animation_s *ss_anim, int32_t ver)

Reply:
i will change it fast :+1: 
for define it just to view what sound is in int[] like "result[SHOT] here is shot_snd" !
but can be removed if you dont want it and i not have collision in-game for now ;x

Reply:
returning unnamed int* with access by half magic access by offset like SHOT is not good (developer must know where search defines and check them for "is it's usage correct?").